packages/fixie/src/agent.ts

```diff
@@ -480,7 +488,19 @@ export class FixieAgent {
     this.getCodePackage(agentPath);
 
     // Start the agent process locally.
-    FixieAgent.spawnAgentProcess(agentPath, port, environmentVariables);
+    let agentProcess = FixieAgent.spawnAgentProcess(agentPath, port, environmentVariables);
+
+    // Watch files in the agent directory for changes.
+    const watchPath = path.resolve(agentPath);
+    console.log(`🌱 Watching ${watchPath} for changes...`);
+    const watcher = new Watcher(watchPath, { ignoreInitial: true, recursive: true });
+    watcher.on('all', async (event: any, targetPath: string, _targetPathNext: any) => {
+      console.log(`🌱 Restarting local agent process due to ${event}: ${targetPath}`);
+      agentProcess.kill();
+      // Let it shut down gracefully.
+      await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Hmm, so this means every agent reload is 2 seconds? Web devs complain if their hot reload is longer than 250ms.

When I did this for the v1 SDK, I had the server process do the reload itself, so there was no need to kill/respawn a subproc: https://github.com/fixie-ai/fixie-sdk/blob/feature-ts/fixieai/agents-ts/src/serve.ts#L169-L178. Could that work here as well?